Ginger/GingerCoreNETUnitTest/RunTestslib/CLITest.cs

```diff
@@ -218,6 +217,156 @@ public void CLIDynamicTest()
             Assert.AreEqual(WorkSpace.Instance.RunsetExecutor.Runners[0].BusinessFlows[0].RunStatus, eRunStatus.Passed, "BF RunStatus=Passed");
         }
 
+
+        /// <summary>
+        /// Testing JSON config creation and execution
+        /// </summary>
+        [TestMethod]
+        public void CLIDynamicJSON_CreateAndExecute_Test()
+        {
+            // Arrange
+            PrepareForCLICreationAndExecution(runsetName:"Calc_Test");
+            // Create config file
+            CLIHelper cLIHelper = new CLIHelper();
+            cLIHelper.RunAnalyzer = true;
+            cLIHelper.ShowAutoRunWindow = false;
+            cLIHelper.DownloadUpgradeSolutionFromSourceControl = false;
+            
+            RunSetAutoRunConfiguration runSetAutoRunConfiguration = new RunSetAutoRunConfiguration(WorkSpace.Instance.Solution, WorkSpace.Instance.RunsetExecutor, cLIHelper);
+            runSetAutoRunConfiguration.ConfigFileFolderPath = mTempFolder;
+            runSetAutoRunConfiguration.SelectedCLI = new CLIDynamicFile(CLIDynamicFile.eFileType.JSON);
+            runSetAutoRunConfiguration.CreateContentFile();
+
+            // Act            
+            CLIProcessor CLI = new CLIProcessor();
+            CLI.ExecuteArgs(new string[] { "dynamic", "-f", runSetAutoRunConfiguration.ConfigFileFullPath });
+
+            // Assert        
+            Assert.AreEqual(WorkSpace.Instance.RunsetExecutor.RunSetConfig.Name, "Calc_Test", "Validating correct Run set was executed" );
+            Assert.AreEqual(WorkSpace.Instance.RunsetExecutor.Runners[0].BusinessFlows[0].Name, "Calculator_Test", "Validating correct Business Flow was executed");
+            Assert.AreEqual(WorkSpace.Instance.RunsetExecutor.Runners[0].BusinessFlows[0].RunStatus, eRunStatus.Passed, "Validating BF execution Passed");
+        }
+
+        /// <summary>
+        /// Testing JSON existing Runset with customized Values execution
+        /// </summary>
+        [Ignore]
+        [TestMethod]
+        public void CLIDynamicJSON_ExistingCustomized_IDsAndNames_Test()
+        {
+            // Arrange
+
+            // Act            
+            CLIProcessor CLI = new CLIProcessor();
+            CLI.ExecuteArgs(new string[] { "dynamic", "-f", Path.Combine(TestResources.GetTestResourcesFolder("CLI"), "CLI-CustomExistingRunset.Ginger.AutoRunConfigs.json") });      
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4930360)